src/analysis/stack-lattice.h

```diff
@@ -0,0 +1,187 @@
+#ifndef wasm_analysis_stack_lattice_h
+#define wasm_analysis_stack_lattice_h
+
+#include <deque>
+#include <optional>
+
+#include "lattice.h"
+
+namespace wasm::analysis {
+
+// Models the WASM value stack
+// A lattice for modelling stack-related phenomena. It is templated
+// on a lattice representing the contents of each stack element.
+template<typename StackElementLattice> class StackLattice {
+  static_assert(is_lattice<StackElementLattice>);
+  StackElementLattice& stackElementLattice;
+
+public:
+  StackLattice(StackElementLattice& stackElementLattice)
+    : stackElementLattice(stackElementLattice) {}
+
+  class Element {
+    // The top lattice is
```

Comment:
Is this comment cut off?

Reply:
No, I'm not done writing the comments yet.